src/marqo/tensor_search/validation.py

```diff
@@ -501,7 +510,66 @@ def validate_multimodal_combination(field_content, is_non_tensor_field, field_ma
             f"It must be a tensor field. Add this field to `tensor_fields` or"
             f"add it as a normal field to fix this problem."
         )
-    return True
+    return field_content
+
+
+def validate_custom_vector(field_content: dict, is_non_tensor_field: bool, index_model_dimensions: int):
+    '''
+    Validates the field content if it is a custom vector field (dict)
+    Args:
+        field_content: the field content
+        is_non_tensor_field: whether this is a non-tensor-field
+        index_model_dimensions: the `dimensions` property of the index to be added to
+
+    Returns:
+        field_content if the validation passes
+        "content" key will be added to the field_content as empty string if it is not provided.
+    '''
+    if index_model_dimensions is None:
+        raise InternalError(
+            f"Index model dimensions should not be `None`."
+        )
+    
+    # Must be a tensor_field
+    if is_non_tensor_field:
+        raise InvalidArgError(
+            f"Cannot create custom_vector field (given field content: `{field_content}`) as a non-tensor field. "
+            f"Add this field to `tensor_fields` to fix this problem."
+        )
+    
+    try:
+        custom_vector_schema = {
```

Comment:
I would prefer not to, since this is validation for the custom vector object, not its mapping. The custom vector mapping schema is already in the `models.mappings_object`

Reply:
Done. moved to `models.custom_vector_object.py`